libflux/src/flux/semantic/builtins.rs

```diff
@@ -301,6 +301,178 @@ pub fn builtins() -> Builtins<'static> {
+                        ?columns: [string]
+                    ) -> [t1]
+                "#),
+                "derivative" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?unit: duration,
+                        ?nonNegative: bool,
+                        ?columns: [string],
+                        ?timeColumn: string
+                    ) -> [t1]
+                "#),
+                "difference" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?nonNegative: bool,
+                        ?columns: [string],
+                        ?keepFirst: bool
+                    ) -> [t1]
+                "#),
+                "distinct" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "drop" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t1: Row, t2: Row (
+                        <-tables: [t0],
+                        ?fn: (column: string) -> bool,
+                        ?columns: t1
+                    ) -> [t2]
+                "#),
+                "duplicate" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        column: string,
+                        as: string
+                    ) -> [t1]
+                "#),
+                "duration" => Node::Builtin("forall [t0] (v: t0) -> duration"),
+                "elapsed" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?unit: duration,
+                        ?timeColumn: string,
+                        ?columnName: string
+                    ) -> [t1]
+                "#),
+                "exponentialMovingAverage" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        n: int
+                    ) -> [t1]
+                "#),
+                "false" => Node::Builtin("forall [] () -> bool"),
+                "fill" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        value: [t1],
+                        usePrevious: bool
+                    ) -> [t2]
+                "#),
+                "filter" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        fn: (r: t1) -> bool
+                    ) -> [t2]
+                "#),
```

Comment:
Note, `filter` takes in a list of records and returns that same list of records (with some elements possibly removed). Therefore because it's not modifying the input type in any way before it outputs, the type signature only needs a single parameter, like so:

    forall [t0] where t0: Row (<-tables: [t0], fn: (r: t0) -> bool) -> [t0]